src/claimtrie.cpp

```diff
@@ -137,6 +149,75 @@ void CClaimTrieNode::reorderClaims(supportMapEntryType& supports)
+        if (normalizationForkStateInstance.nonNormalizedTrie == NULL) {
+            normalizationForkStateInstance.nonNormalizedTrie = this;
+            normalizationForkStateInstance.normalizedTrie = new CClaimTrie(false, false, true);
+            normalizationForkStateInstance.normalizedTrie->ReadFromDisk(true);
+            normalizationForkStateInstance.normalizedTrie->nCurrentHeight = nHeight;
+            normalizationForkStateInstance.normalizedTrie->copyAndNormalizeQueues(*this);
+
+            nodes = normalizationForkStateInstance.normalizedTrie->flattenTrie();
+            LogPrintf("Post-normalization fork claim trie nodes = %lu\n", nodes.size());
+            // Debug-only
+            /* for (std::vector<namedNodeType>::iterator it = nodes.begin(); it != nodes.end(); ++it) */
+            /* { */
+            /*     if (!it->second.claims.empty()) */
+            /*         LogPrintf("[***] Claim name: \"%s\" (hash %s)\n", it->first.c_str(), it->second.hash.GetHex().c_str()); */
+            /*     else */
+            /*         LogPrintf("[EMPTY] Claim name: \"%s\" (hash %s)\n", it->first.c_str(), it->second.hash.GetHex().c_str()); */
+            /* } */
+        }
+
+        assert(normalizationForkStateInstance.nonNormalizedTrie);
+        return normalizationForkStateInstance.normalizedTrie;
+    }
+
+    // NOTE: Do not call this with enable == false before enabling it
+    assert(normalizationForkStateInstance.nonNormalizedTrie);
+    return normalizationForkStateInstance.nonNormalizedTrie;
+}
+
+void CClaimTrie::copyAndNormalizeQueues(const CClaimTrie& other)
+{
+    db = other.db;
+    nCurrentHeight = other.nCurrentHeight;
+    nExpirationTime = other.nExpirationTime;
+    nProportionalDelayFactor = other.nProportionalDelayFactor;
+    root = other.root;
+    hashBlock = other.hashBlock;
+    for (claimQueueType::const_iterator it = other.dirtyQueueRows.begin(); it != other.dirtyQueueRows.end(); ++it)
+        dirtyQueueRows[it->first] = it->second;
+    for (queueNameType::const_iterator it = other.dirtyQueueNameRows.begin(); it != other.dirtyQueueNameRows.end(); ++it)
+        dirtyQueueNameRows[normalizeClaimName(it->first)] = it->second;
+    for (expirationQueueType::const_iterator it = other.dirtyExpirationQueueRows.begin(); it != other.dirtyExpirationQueueRows.end(); ++it)
+        dirtyExpirationQueueRows[it->first] = it->second;
+    for (supportQueueType::const_iterator it = other.dirtySupportQueueRows.begin(); it != other.dirtySupportQueueRows.end(); ++it)
+        dirtySupportQueueRows[it->first] = it->second;
+    for (expirationQueueType::const_iterator it = other.dirtySupportExpirationQueueRows.begin(); it != other.dirtySupportExpirationQueueRows.end(); ++it)
+        dirtySupportExpirationQueueRows[it->first] = it->second;
+    for (nodeCacheType::const_iterator it = other.dirtyNodes.begin(); it != other.dirtyNodes.end(); ++it)
+        dirtyNodes[normalizeClaimName(it->first)] = new CClaimTrieNode();
```

Comment:
Does we need to copy trie node?
```
dirtyNodes[normalizeClaimName(it->first)] = new CClaimTrieNode(*(it->second));
```

Reply:
But why we want an empty node, it can wipe db in some cases.

Reply:
Normalization can narrow the claimtrie name space.

Reply:
What if the existing node had additional claims on it?